tests/foreman/ui/test_registration.py

```diff
@@ -0,0 +1,337 @@
+
+
+@pytest.mark.tier3
+@pytest.mark.usefixtures('enable_capsule_for_registration')
+def test_global_re_registration_host_with_force_ignore_error_options(
+    session, module_activation_key, default_os, default_smart_proxy, rhel7_contenthost
+):
+    """If the ignore_error and force checkbox is checked then registered host can
+    get re-registered without any error.
+
+    :id: 8f0ecc13-5d18-4adb-acf5-3f3276dccbb7
+
+    :expectedresults: Verify the force and ignore checkbox options
+
+    :CaseLevel: Integration
+
+    :steps:
+        1. create and sync repository
+        2. create the content view and activation-key
+        3. open the global registration form and select --force and --Ignore Errors option
+        4. registered the host with generated curl command
+        5. re-register the same host again and check it is getting registered
+
+    :parametrized: yes
+    """
+    client = rhel7_contenthost
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.capsule': default_smart_proxy.name,
+                'general.activation_keys': module_activation_key.name,
+                'general.insecure': True,
+                'advanced.force': True,
+                'advanced.ignore_error': True,
+            }
+        )
+    client.execute(cmd)
+    result = client.execute('subscription-manager identity')
+    assert result.status == 0
+    # rerun the register command
+    client.execute(cmd)
+    result = client.execute('subscription-manager identity')
+    assert result.status == 0
+
+
+@pytest.mark.rhel_ver_match('8')
+def test_host_facts_upload_on_host_registration(
+    session,
+    module_target_sat,
+    module_activation_key,
+    rhel_contenthost,
+    default_os,
+    default_smart_proxy,
+    module_entitlement_manifest_org,
+):
+    """Host Facts are uploaded to satellite when host is registered using global registration form.
+
+    :id: fe16d66a-8e0b-4872-a6bf-36c5c12d8d5c
+
+    :CaseLevel: Integration
+
+    :steps:
+        1. Enable and sync any repository.
+        2. Create CV, add synced repo and publish it to library environment.
+        3. Create an activation key
+        4. Register host using global registration curl command.
+        5. Navigate to Hosts -> All Hosts -> Hostname
+
+    :expectedresults: Verify that host facts are uploaded to satellite when host
+        is registered using global registration
+
+    :BZ: 2001552
+
+    :CaseAutomation: Automated
+
+    :customerscenario: true
+
+    :CaseImportance: High
+
+    :parametrized: yes
+    """
+    module_target_sat.api_factory.enable_sync_redhat_repo(
+        constants.REPOS['rhel8_bos'], module_entitlement_manifest_org.id
+    )
+    cv = module_target_sat.api.ContentView(organization=module_entitlement_manifest_org).create()
+    cv.publish()
+    with module_target_sat.ui_session() as session:
```

Comment:
Why are you not consistent with using `session` and `target_sat.ui_session()` in this test module? I would recommend using the latter one.